test/headersReceived.test.js

```diff
@@ -109,7 +114,110 @@ describe("check redemption attempt conditions", () => {
+        beforeEach(() => {
+            getMock = function() { return 2; };
+            workflow.__set__("get", getMock);
+        });
+
+        test("not set", () => {
+            workflow.__set__("SPEND_IFRAME", false);
+            let fired = processHeaders(details, url);
+            expect(fired).toBeTruthy;
+            expect(updateIconMock).toBeCalledTimes(3);
+        });
+
+        test("set and iframe", () => {
+            workflow.__set__("SPEND_IFRAME", true);
+            workflow.__set__("iframe", true);
+            let fired = processHeaders(details, url);
+            expect(fired).toBeTruthy;
+            expect(updateIconMock).toBeCalledTimes(3);
+        });
+
+        test("set and not iframe", () => {
+            workflow.__set__("SPEND_IFRAME", true);
+            workflow.__set__("iframe", false);
+            let fired = processHeaders(details, url);
+            expect(fired).toBeTruthy;
+            expect(updateIconMock).not.toBeCalled;
+        });
+    });
+
+    describe("setting of readySign", () => {
+        beforeEach(() => {
+            getMock = function() { return 0; };
+            workflow.__set__("get", getMock);
+        });
+
+        describe("signing enabled", () => {
+            beforeEach(() => {
+                workflow.__set__("DO_SIGN", true);
+                workflow.__set__("readySign", false);
+            });
+
+            test("no tokens", () => {
+                let fired = processHeaders(details, url);
+                expect(fired).toBeFalsy();
+                let readySign = workflow.__get__("readySign");
+                expect(readySign).toBeTruthy();
+            });
+
+            test("not activated", () => {
+                header = { name: "Different-header-name", value: "1" };
+                details.responseHeaders = [header];
+                let fired = processHeaders(details, url);
+                expect(fired).toBeFalsy();
+                let readySign = workflow.__get__("readySign");
+                expect(readySign).toBeFalsy();
+            });
+
+            test("tokens > 0", () => {
+                getMock = function() { return 2; };
+                workflow.__set__("get", getMock);
+                let fired = processHeaders(details, url);
+                expect(fired).toBeTruthy();
+                let readySign = workflow.__get__("readySign");
+                expect(readySign).toBeFalsy();
+            });
+
+            test("tokens > 0 but captcha.website", () => {
+                url = new URL("https://captcha.website");
+                getMock = function() { return 2; };
+                workflow.__set__("get", getMock);
+                let fired = processHeaders(details, url);
+                expect(fired).toBeFalsy();
+                let readySign = workflow.__get__("readySign");
+                expect(readySign).toBeTruthy();
+            });
+
+            test("redemption off", () => {
```

Comment:
Maybe add one more unit test for background.js:168 (that is, checking that the button icon becomes (!) when token count == 0

Reply:
Good call, I've added an extra line to the test "no tokens" (headersReceived.test.js:167) that checks whether the updateIcon method was called with "!".